mftf/2.3/reporting.md

```diff
@@ -0,0 +1,367 @@
+- Total results of the test run such as number of tests, assertions, and failures.
+
+To manage the level of verbosity, use `-v` or `--verbose` flag in the `mftf` commands.
+To enable verbosity using the `codecept` commands, refer to the Codeception [Console Commands][codeception].
+
+The following sections demonstrate an example interpretation of a complete log separated into chunks.
+
+### Pre-run check report
+
+First, the MFTF reports about issues with environment.
+In our case, there is an issue with PHP library loading.
+
+```terminal
+PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/local/lib/php/pecl/20160303/php_intl.dll' - dlopen(/usr/local/lib/php/pecl/20160303/php_intl.dll, 9): image not found in Unknown on line 0
+```
+{: .no-copy }
+
+Next, the MFTF returns `DEPRECATION` reports alerting you that required test components are missing in XML test definitions.
+
+```terminal
+DEPRECATION: Test AdminFilteringCategoryProductsUsingScopeSelectorTest is missing required annotations.{"testName":"AdminFilteringCategoryProductsUsingScopeSelectorTest","missingAnnotations":"stories"}
+DEPRECATION: Test AdminAbleToShipPartiallyInvoicedItemsTest is missing required annotations.{"testName":"AdminAbleToShipPartiallyInvoicedItemsTest","missingAnnotations":"stories"}
+DEPRECATION: Test AdminRemoveProductWeeeAttributeOptionTest is missing required annotations.{"testName":"AdminRemoveProductWeeeAttributeOptionTest","missingAnnotations":"stories"}
+Generate Tests Command Run
+```
+{: .no-copy }
+
+`Generate Tests Command Run` indicates the moment when the MFTF has run the tests generation command actually.
+
+### Test execution report
+
+A test execution report is generated by Codeception and includes configuration information, scenario execution steps, and PASSED/FAIL verdict after each test completion.
+
+#### General information
+
+The general information can be useful for MFTF contributors, but can be ignored by a test writer.
+
+Let's consider the general part of the following test execution report:
+
+```terminal
+==== Redirecting to Composer-installed version in vendor/codeception ====
+Codeception PHP Testing Framework v2.3.9
+Powered by PHPUnit 6.5.13 by Sebastian Bergmann and contributors.
+
+Magento\FunctionalTestingFramework.functional Tests (2) ------------------------
+Modules: \Magento\FunctionalTestingFramework\Module\MagentoWebDriver, \Magento\FunctionalTestingFramework\Helper\Acceptance, \Magento\FunctionalTestingFramework\Helper\MagentoFakerData, \Magento\FunctionalTestingFramework\Module\MagentoRestDriver, PhpBrowser, \Magento\FunctionalTestingFramework\Module\MagentoSequence, \Magento\FunctionalTes
+```
+
+After the test generation command (mentioned in the previous section), the MFTF delegates control to the `vendor/codeception` tool, which is the `Codeception PHP Testing Framework` of version `2.3.9` that uses `PHPUnit` of version `6.5.13`.
```

Comment:
Are version numbers necessary here? Just want to make sure since this will be added to future changes as a place to check.

Reply:
Yes, it is just interpretation of the report above.
There is no need to always keep the latest available versions here.